backend/json.go

```diff
@@ -149,10 +149,24 @@ func readQueryDataResultsJSON(qdr *QueryDataResponse, iter *jsoniter.Iterator) {
 
 			qdr.Responses = make(Responses)
 
-			for l2Field := iter.ReadObject(); l2Field != ""; l2Field = iter.ReadObject() {
+			l2Field := iter.ReadObject()
+			for {
+				// the response may have an empty-string refId. this is not allowed,
+				// but it may happen, and we want to be able to handle the case.
```

Comment:
>but it may happen, and we want to be able to handle the case.

Out of curiosity, how can this currently happen? Do we need validation elsewhere?

Reply:
+1, we are removing the error here, therefore making having an empty refId valid. Is that what we want? From the issue description it appears that the goal is to return a meaningful error message, not remove the error.

Reply:
Clarified offline that we want to actually avoid the error here.